src/wolfio.c

```diff
@@ -133,31 +133,46 @@ int BioReceive(WOLFSSL* ssl, char* buf, int sz, void* ctx)
         return WOLFSSL_CBIO_ERR_GENERAL;
     }
 
-    if (ssl->biord->method && ssl->biord->method->readCb) {
-        WOLFSSL_MSG("Calling custom biord");
-        recvd = ssl->biord->method->readCb(ssl->biord, buf, sz);
-        if (recvd < 0 && recvd != WOLFSSL_CBIO_ERR_WANT_READ)
-            return WOLFSSL_CBIO_ERR_GENERAL;
-        return recvd;
-    }
+    if (wolfSSL_BIO_supports_pending(ssl->biord) &&
```

Comment:
Old code tries to perform read callback first then checks for pending. New code does opposite, so the callback never gets a chance to run to actually read data.

Reply:
Its enough to move the pending check inside the error handling of the read operation. I didn't consider that data could be acquired during the read call. Can you take it or do you want me to fix this on Monday?